src/main/java/rx/internal/operators/OperatorSwitchIfEmpty.java

```diff
@@ -38,33 +38,35 @@ public OperatorSwitchIfEmpty(Observable<? extends T> alternate) {
         final SwitchIfEmptySubscriber parent = new SwitchIfEmptySubscriber(child, ssub);
         ssub.set(parent);
         child.add(ssub);
+        child.setProducer(new Producer() {
```

Comment:
Yeah life would be easier if `request()` was public. I wouldn't mind seeing it opened up, do you think it worth discussing in an issue?